### PR TITLE
delete vertex from dict type inclist

### PR DESCRIPTION
**Pull request by dehann**
don't merge yet,  requesting additional testing from @GearsAD please -- regarding request for delete_vertex! function.

### CODE REVIEW

**Comment by GearsAD** [comment]
Great, thanks - will look this afternoon

File: src/incidence_list.jl [hunk 1]
```diff
@@ -110,6 +110,41 @@ function add_vertex!(g::GenericIncidenceList{V,E}, v::V) where {V,E}
...
+      end
+    end
+    delete!(inclist, vid)
+  end
+
+  delete!(vertices, v.index)
+  return nedges
+end
+
+function delete_vertex!(g::GenericIncidenceList{V,E}, v::V) where {V,E}
```
Comment (GearsAD) [inline src/incidence_list.jl]: Suggestion - standard pattern would be delete_vertex(v::V, g::GenericIncidenceList{V,E}) where {V,E})

Maybe change to match standard?

**Comment by GearsAD** [comment]
Additional testing for tests/inclist.jl.

The end of the test seems to be failing, to be dicussed - why does the graph break when I delete variables[4]?

```julia
@testset "Incidence Dictionaries" do

g = Graphs.incdict(Graphs.ExVertex,is_directed=false)
variables = [ExVertex(1, "x1"), ExVertex(2, "x2"), ExVertex(3, "x3"), ExVertex(4, "x4")]
map(v -> add_vertex!(g, v), variables)

for i = 1 : length(variables)
    @test vertices(g)[i] in variables
    @test out_degree(variables[i], g) == 0
end

factors =[ExVertex(5, "x1x2f1"), ExVertex(6, "x2x3f1"), ExVertex(7, "x3x4f1")]
add_vertex!(g, factors[1])
edge = Graphs.make_edge(g, variables[1], factors[1]); Graphs.add_edge!(g, edge)
edge = Graphs.make_edge(g, factors[1], variables[2]); Graphs.add_edge!(g, edge)
add_vertex!(g, factors[2])
edge = Graphs.make_edge(g, variables[2], factors[2]); Graphs.add_edge!(g, edge)
edge = Graphs.make_edge(g, factors[2], variables[3]); Graphs.add_edge!(g, edge)
add_vertex!(g, factors[3])
edge = Graphs.make_edge(g, variables[3], factors[3]); Graphs.add_edge!(g, edge)
edge = Graphs.make_edge(g, factors[3], variables[4]); Graphs.add_edge!(g, edge)

@test num_vertices(g) == 7

@test [out_degree(v, g) for v in variables] == [1, 2, 2, 1]
@test [out_degree(f, g) for f in factors] == [2, 2, 2]

@test setdiff(collect(out_neighbors(variables[1], g)), [factors[1]]) == []
@test setdiff(collect(out_neighbors(variables[2], g)), [factors[1], factors[2]]) == []
@test setdiff(collect(out_neighbors(variables[3], g)), [factors[2], factors[3]]) == []
@test setdiff(collect(out_neighbors(variables[4], g)), [factors[3]]) == []
@test setdiff(collect(in_neighbors(variables[1], g)), [factors[1]]) == []
@test setdiff(collect(in_neighbors(variables[2], g)), [factors[1], factors[2]]) == []
@test setdiff(collect(in_neighbors(variables[3], g)), [factors[2], factors[3]]) == []
@test collect(in_neighbors(variables[4], g)) == [factors[3]]

# Delete testing
delete_vertex!(g, variables[4])
@test num_vertices(g) == 6
@test_throws Exception out_degree(variables[4], g)
@test out_degree(factors[3], g) == 1 # Should be just x3 now
# Discuss here - why is this failing?
end
```

**Comment by dehann** [comment]
Ah thanks!  will have a look and add the tests

File: src/incidence_list.jl (hunk 1 above)
Comment (dehann) [reply src/incidence_list.jl]: done

**Pull request merged**